### PR TITLE
Use unique Oracle DB TCP listener port for every DB creation test run

### PR DESCRIPTION
**Pull request by maggiehe00**
In this PR i used getNextFreePort() to get unique port for Oracle DB TCP listener port for DB service/pod creation and keep DB service targetPort 1521, since 1521 is hardcoded in the DB image as DB end port.
Sequential test passed at 
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/5934/
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/5935/

Parallel test without DB creation issue at
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/5936/
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/5938/
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/5939/
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/5941/

After merging to the latest main,  sequential passed at
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/5962
No DB creation related failure in parallel run
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/5974/

### CODE REVIEW

**Review by hzhao-github — APPROVED** [review]

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItCrossDomainTransaction.java [hunk 1]
```diff
@@ -113,7 +113,7 @@
   private static String domain2AdminServerPodName = domainUid2 + "-admin-server";
   private final String domain2ManagedServerPrefix = domainUid2 + "-managed-server";
   private static final String ORACLEDBURLPREFIX = "oracledb.";
-  private static final String ORACLEDBSUFFIX = ".svc.cluster.local:1521/devpdb.k8s";
+  private static String ORACLEDBSUFFIX = null;
   private static LoggingFacade logger = null;
   static String dbUrl;
   static int dbNodePort;
```
[the remark this answers is not in the document]
Comment (hzhao-github) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItCrossDomainTransaction.java:119]: I guess this is not from your change :)

[the remark this answers is not in the document]
Comment (hzhao-github) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItCrossDomainTransaction.java:119]: LGTM

[the remark this answers is not in the document]
Comment (maggiehe00) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItCrossDomainTransaction.java:119]: Which 2 variables are you referring to? dbUrl and dbNodePort?

**Review by sankarpn — APPROVED** [review]
lgtm

**Pull request merged**